### DISCUSSION TITLE
Update the title of a scalar bar

### DISCUSSION THREAD

**Discussion by p-j-smith** [pyvista/pyvista#2272] [Q&A] [answered]
Hi, I would like to update the title of a scalar bar when changing the active scalars of the corresponding mesh. I'm wondering whether it's possible to this using pyvista?

I can update the title like this:

```py
plotter.renderer._actors['Addr=0x7fc734799d90'].SetTitle('new title')
```

but this requires me to know the memory address of the scalar bar.

Alternatively, is it possible to give the scalar bar a name when creating it so that it could be accessed like: `plotter.renderer._actors['My scalar bar']`?

**Comment by banesullivan** [comment, marked as answer]
Updating after the fact like this hasn't yet been implemented. Another workaround would be to use the `add_scalar_bar` method directly:

```py
import pyvista as pv
mesh = pv.Wavelet()

p = pv.Plotter()
actor = p.add_mesh(mesh, show_scalar_bar=False)

scalar_bar_actor = p.add_scalar_bar(title='Original Title', mapper=actor.GetMapper())

scalar_bar_actor.SetTitle('New Title')
p.show()
```
![Screen Shot 2022-06-27 at 5 54 08 PM](https://user-images.githubusercontent.com/22067021/176058471-486901b8-ca87-4354-b8de-21515c156781.png)

**Reply by p-j-smith** [reply]
That's a good idea, thank you!
